crates/syn2mas/src/migration.rs

```diff
@@ -132,7 +132,8 @@ async fn migrate_users(
     server_name: &str,
     rng: &mut impl RngCore,
 ) -> Result<UsersMigrated, Error> {
-    let mut write_buffer = MasUserWriteBuffer::new(mas);
+    let mut user_buffer = MasWriteBuffer::new(MasWriter::write_users);
```

Comment:
not sure how I feel about passing function pointers instead of using traits… I guess that works, not sure how it would look like with traits